build_tools/build_detect_platform

```diff
@@ -317,9 +317,8 @@ EOF
     # Test whether jemalloc is available
     if echo 'int main() {}' | $CXX $CFLAGS -x c++ - -o /dev/null -ljemalloc \
       2>/dev/null; then
-        PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -ljemalloc"
-        JAVA_LDFLAGS="$JAVA_LDFLAGS -ljemalloc"
         JEMALLOC=1
+        WITH_JEMALLOC_FLAG=1
```

Comment:
@sagar0 I hope it is more clear after the comments. I think JEMALLOC could be renamed to USE_ JEMALLOC_RELATED_IDS or even better removed entirely and being inferred from WITH_JEMALLOC_FLAG and JEMALLOC_LIB but for backward compatibility we are keeping it.